Modules/Registration/Common/include/itkMattesMutualInformationImageToImageMetric.h

```diff
@@ -323,10 +323,6 @@ class ITK_TEMPLATE_EXPORT MattesMutualInformationImageToImageMetric
   PDFValueType  m_FixedImageBinSize{ 0.0 };
   PDFValueType  m_MovingImageBinSize{ 0.0 };
 
-  /** Cubic BSpline kernel for computing Parzen histograms. */
-  typename CubicBSplineFunctionType::Pointer           m_CubicBSplineKernel;
```

Comment:
Do we need to guard this under `legacy` ifdef?

Reply:
Both "cubic BSpline kernel" data members of `MattesMutualInformationImageToImageMetric` were private, and neither of them had getters or setters. So I don't think there is any possible reason to guard them under legacy ifdef. 

The corresponding "cubic BSpline kernel" data members of the v4 version (`MattesMutualInformationImageToImageMetricv4`) were not part of the public interface either. These two were declared `protected`, but I think they should also have been declared `private` in the first place. They were only accessed by its friend class, `MattesMutualInformationImageToImageMetricv4GetValueAndDerivativeThreader`. So I think they can also be removed without legacy support. Is that OK to you?

Reply:
If they were not public, no need to put them through legacy process.